pkg/infrastructure/services/deployment.go

```diff
@@ -103,3 +105,12 @@ func setHTTPPortInEnvVar(httpPort int32, kogitoService v1alpha1.KogitoService) {
 	modifiedEnv := framework.EnvOverride(envs, httpPortEnvVar)
 	kogitoService.GetSpec().SetEnvs(modifiedEnv)
 }
+
+func isDeploymentAvailable(cli *client.Client, name, namespace string) (bool, error) {
```

Comment:
will this be called only during reconciliation?

Reply:
yes @spolti . We call this function while configuring prometheus as pre-check. Operator goes to configure prom only when deployment is ready to serve request because operator have to application's `/metrics`  endpoint to check whether application support prom or not

https://github.com/kiegroup/kogito-cloud-operator/blob/640c94390db7cdef2f2e95f258c51013195ebae5/pkg/infrastructure/services/prometheus.go#L42